Library/Formula/openssl.rb

```diff
@@ -72,7 +66,13 @@ def install
         system "make", "clean"
       end
 
-      ENV.deparallelize
+
+      # This ensures where Homebrew's Perl is needed the Cellar path isn't
+      # hardcoded into OpenSSL's scripts, causing them to break every Perl update.
+      # Whilst our env points to opt_bin, by default OpenSSL resolves the symlink.
+      ENV["PERL"] = Formula["perl"].opt_bin/"perl" if which("perl") == Formula["perl"].opt_bin/"perl"
```

Comment:
Isn't your `if` statement always true here given that `perl` is a build dependency?